From 4fd644bda360fa9853c950962fb413bd58e1e472 Mon Sep 17 00:00:00 2001
From: John Andersen <john.s.andersen@intel.com>
Date: Fri, 15 Nov 2019 13:25:37 -0500
Subject: [PATCH 4/6] KVM: X86: ops: Add setters for guest owned CR bits

Add pointers to kvm_x86_ops to allow access to vmx and svm setters for
guest owned CR bits.

Signed-off-by: John Andersen <john.s.andersen@intel.com>
---
 arch/x86/include/asm/kvm_host.h | 4 ++++
 arch/x86/kvm/svm.c              | 2 ++
 arch/x86/kvm/vmx/vmx.c          | 2 ++
 3 files changed, 8 insertions(+)

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 4fc61483919a..ab8550182a21 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1046,6 +1046,10 @@ struct kvm_x86_ops {
 	void (*set_cr0)(struct kvm_vcpu *vcpu, unsigned long cr0);
 	void (*set_cr3)(struct kvm_vcpu *vcpu, unsigned long cr3);
 	int (*set_cr4)(struct kvm_vcpu *vcpu, unsigned long cr4);
+	void (*set_cr0_guest_owned_bits)(struct kvm_vcpu *vcpu,
+			unsigned long cr0_guest_owned_bits);
+	void (*set_cr4_guest_owned_bits)(struct kvm_vcpu *vcpu,
+			unsigned long cr4_guest_owned_bits);
 	void (*set_efer)(struct kvm_vcpu *vcpu, u64 efer);
 	void (*get_idt)(struct kvm_vcpu *vcpu, struct desc_ptr *dt);
 	void (*set_idt)(struct kvm_vcpu *vcpu, struct desc_ptr *dt);
diff --git a/arch/x86/kvm/svm.c b/arch/x86/kvm/svm.c
index d965884be021..17a4ad0be69a 100644
--- a/arch/x86/kvm/svm.c
+++ b/arch/x86/kvm/svm.c
@@ -7231,6 +7231,8 @@ static struct kvm_x86_ops svm_x86_ops __ro_after_init = {
 	.set_cr0 = svm_set_cr0,
 	.set_cr3 = svm_set_cr3,
 	.set_cr4 = svm_set_cr4,
+	.set_cr0_guest_owned_bits = svm_set_cr0_guest_owned_bits,
+	.set_cr4_guest_owned_bits = svm_set_cr4_guest_owned_bits,
 	.set_efer = svm_set_efer,
 	.get_idt = svm_get_idt,
 	.set_idt = svm_set_idt,
diff --git a/arch/x86/kvm/vmx/vmx.c b/arch/x86/kvm/vmx/vmx.c
index 9cde415ba875..017661898268 100644
--- a/arch/x86/kvm/vmx/vmx.c
+++ b/arch/x86/kvm/vmx/vmx.c
@@ -7790,6 +7790,8 @@ static struct kvm_x86_ops vmx_x86_ops __ro_after_init = {
 	.set_cr0 = vmx_set_cr0,
 	.set_cr3 = vmx_set_cr3,
 	.set_cr4 = vmx_set_cr4,
+	.set_cr0_guest_owned_bits = vmx_set_cr0_guest_owned_bits,
+	.set_cr4_guest_owned_bits = vmx_set_cr4_guest_owned_bits,
 	.set_efer = vmx_set_efer,
 	.get_idt = vmx_get_idt,
 	.set_idt = vmx_set_idt,
-- 
2.21.0

